src/format/scan.rs

```diff
@@ -15,6 +15,16 @@ use crate::Weekday;
 /// Any number that does not fit in `i64` is an error.
 #[inline]
 pub(super) fn number(s: &str, min: usize, max: usize) -> ParseResult<(&str, i64)> {
+    let (s, n) = negative_number(s, min, max)?;
```

Comment:
I'm very confused by the logic of parsing a non-negative number by first parsing a negative number and then negating it...

Reply:
I thought it was quite nice. Positive numbers are not 'better' than negative numbers, and in this case negative numbers had the one integer larger range we needed.

Updated to parse to an `u64` and convert with a tricky cast instead.